src/main/java/frc/robot/subsystems/FieldMap.java

```diff
@@ -1,40 +1,321 @@
-/*----------------------------------------------------------------------------*/
-/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
-/* Open Source Software - may be modified and shared by FRC teams. The code   */
-/* must be accompanied by the FIRST BSD license file in the root directory of */
-/* the project.                                                               */
-/*----------------------------------------------------------------------------*/
-
 package frc.robot.subsystems;
 
-import edu.wpi.first.wpilibj2.command.CommandBase;
-
-public class FieldPosition extends CommandBase {
-  /**
-   * Creates a new FieldPosition.
-   */
-  public FieldPosition() {
-    // Use addRequirements() here to declare subsystem dependencies.
-  }
-
-  // Called when the command is initially scheduled.
-  @Override
-  public void initialize() {
-  }
-
-  // Called every time the scheduler runs while the command is scheduled.
-  @Override
-  public void execute() {
-  }
-
-  // Called once the command ends or is interrupted.
-  @Override
-  public void end(boolean interrupted) {
-  }
-
-  // Returns true when the command should end.
-  @Override
-  public boolean isFinished() {
-    return false;
-  }
-}
+
+
+import java.util.ArrayList;
+
+import java.util.LinkedList;
+
+import java.util.List;
+
+import org.ligerbots.powerup.RobotPosition.Action;
+
+import org.ligerbots.powerup.FieldPosition;
+
+
+
+
+public class FieldMap {
+
+    
+
+    
+
+    // Robot dimensions with bumpers
+
+    public static final double robotWidth = 0; // TODO: Determine final robot width and length
+
+    public static final double robotLength = 0;
+
+    // half dimensions - for getting the center point of the robot
+
+    public static final double rW2 = robotWidth/2.0;
+
+    public static final double rL2 = robotLength/2.0;
+
+    public static final double centerLine = 227.42;
+    
+
+    public static FieldPosition[] startPositions = new FieldPosition[5];   // position 0 is not used! (because the diagram is 1-based)
+
+    public static FieldPosition[] ballPosition = new FieldPosition[12];
+
+    public static FieldPosition[] controlPanal = new FieldPosition[0];
```

Comment:
A single element array must be instantiated with a size of 1. This should be:
public static FieldPosition[] controlPanal = new FieldPosition[1];